### DISCUSSION TITLE
Warm-start (restart) option in optimizations

### DISCUSSION THREAD

**Discussion by cuneyd** [mhm-ufz/mHM#5] [Q&A]
Hello,

When the restart option for optimize=TRUE will be actively used? 
Sometimes we have power outage issue and our iterations are lost.

Kind Regards,
Cüneyd

**Comment by lese66** [comment]
Hi Cüneyd,

We use restart files mainly for operation jobs, e.g. monitoring, forecasting or even in climate projections. 
If you have a good parameter set, you can run with interpolated observations until T0, as save a restart file. Then, in another instance, e.g. next days, well forecast forcings are available, read the restart for T0, and move the model in forecast mode until T1. when new observed forcings become available (e,g,E-OBS) you can move the state variables and states until T1. And so on. This is the method we use in the drought monitor www.ufz.de/droughtmonitor or for forecasts. See workflow in https://journals.ametsoc.org/view/journals/bams/100/12/bams-d-17-0274.1.xml
With this procedure you saved enormous computational time because you do not need the warming up of several years and the simulation from T-n... to To. If this is over a continent, this becomes a crucial factor effecting the performance of the forecasting system. 
In climate projections, the historical restart can be create for To, before the ensemble forcings bring the model to T1 ... 100 years in the future...

A climatological restart, I do not recommended for calibration tough. The reason being that parameter sets are biased since historical data on a region was used ti estima them. It is in mHM options because we made experiments with it, aiming to save warming up time during optimization, but we concluded that is not recomendable. Not published however... 

I hope this helps.

Luis

**Comment by StephanThober** [comment]
Hi Cuneyd,

this will never be allowed because the restart file you start from depends on the parameters. The option that you are looking for is to restart the optimisation from a given iteration. You find this in the `mhm.nml` in the `optimize_restart` flag in the `mainconfig_mhm_mrm` namelist.

I hope this helps.

Stephan

**Reply by cuneyd** [reply]
Hi @StephanThober,

The error message below is what I was wondering about (start point of the question) actually.
Except SCE method, we receive this warning: a restart of this opti method is not implemented yet.


optimize = .true.
!> Optimization shall be restarted from ./mo_<opti_method>.restart file, which
!> should be located next to the mhm executable (mhm)
optimize_restart = .true.
!> (0) MCMC                (requires single-objective (SO) function)
!> (1) DDS                 (requires single-objective (SO) function)
!> (2) Simulated Annealing (requires single-objective (SO) function)
!> (3) SCE                 (requires single-objective (SO) function)
!> additional settings for the different methods can be provided below in namelist Optimization
opti_method = 1

![DDS_no_restart](https://user-images.githubusercontent.com/7171087/115850856-fae9a880-a42e-11eb-8a71-5f8021209be3.png)

**Reply by MuellerSeb** [reply]
Hey there,

restarting is only implemented in case of `SCE` (`opti_method=3`).

Writing the restart file is done always when optimization is not started from a given restart file:
```fortran
optimize = .True.
optimize_restart = .False.
```
The `write_restart` will only be used for mHM itself and not for the optimization. File name would be: `'mo_sce.restart'`.

Does this help?
Sebastian

**Reply by cuneyd** [reply]
Hi @MuellerSeb 

I tried these two options with cygwin and ubuntu lts but I couldnt find where "mo_sce.restart" is created.

optimize = .True.
optimize_restart = .False.

I check output_b1 and upper folder. I could only see sce_population and sce_results files. 

![image](https://user-images.githubusercontent.com/7171087/116633956-7c779400-a963-11eb-83dd-02756e1f0432.png)

Cheers,
Cüneyd

**Reply by MuellerSeb** [reply]
I see. Seems to be more crucial issue. Could you create an issue at https://git.ufz.de/mhm/mhm/-/issues ?

**Comment by cuneyd** [comment]
Thank you @lese66 and @StephanThober for the quick and detailed replies.
I found the section in mhm.nml now. I wonder how to create "./mo_<opti_method>.restart" file. 

May be it is created after I set all these three options below as TRUE?
```fortran
write_restart = .True.
optimize = .True.
optimize_restart = .True.
```

------
!> Optimization shall be restarted from ./mo_<opti_method>.restart file, which
!> should be located next to the mhm executable (mhm)
